File: src/pages/BasePage.ts
import { Page } from 'playwright';
import type { EnvironmentConfig } from '../../config/types';

export abstract class BasePage {
  constructor(protected page: Page, protected config: EnvironmentConfig) {}

  async navigate(path: string): Promise<void> {
    await this.page.goto(path);
  }

  async waitForLoad(): Promise<void> {
    await this.page.waitForLoadState('networkidle');
  }

  async getTitle(): Promise<string> {
    return this.page.title();
  }
}

File: java/conductor-core/src/main/java/com/nouhouari/conductor/pages/BasePage.java
package com.nouhouari.conductor.pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.LoadState;
import com.nouhouari.conductor.config.EnvironmentConfig;

/** Java port of src/pages/BasePage.ts. */
public abstract class BasePage {

    protected final Page page;
    protected final EnvironmentConfig config;

    protected BasePage(Page page, EnvironmentConfig config) {
        this.page = page;
        this.config = config;
    }

    public void navigate(String path) {
        page.navigate(path);
    }

    public void waitForLoad() {
        page.waitForLoadState(LoadState.NETWORKIDLE);
    }

    public String getTitle() {
        return page.title();
    }
}
